src/main/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchPushProcessorImpl.java

```diff
@@ -0,0 +1,376 @@
+                    objectStateRecordRepository.save(objectStateRecord);
+                }
+
+                String rawFilename = dataStoreToArmHelper.generateRawFilename(armEod);
+
+                if (dataStoreToArmHelper.shouldPushRawDataToArm(batchItem)) {
+                    pushRawDataAndCreateArchiveRecordIfSuccess(batchItem, rawFilename, userAccount, objectStateRecord);
+                } else if (dataStoreToArmHelper.shouldAddEntryToManifestFile(batchItem)) {
+                    batchItem.setArchiveRecord(archiveRecordService.generateArchiveRecordInfo(batchItem.getArmEod().getId(), rawFilename));
+                }
+            } catch (Exception e) {
+                String errorMessage = String.format("Error during batch push DETS EOD %s to ARM", currentEod.getId());
+                log.error(errorMessage, e);
+                dataStoreToArmHelper.recoverByUpdatingEodToFailedArmStatus(batchItem, userAccount);
+                errorMessage = errorMessage + " - " + ExceptionUtils.getStackTrace(e);
+                updateObjectStateRecordStatus(batchItem.getArmEod(), errorMessage);
+            }
+        }
+
+        if (!writeManifestAndCopyToArm(userAccount, batchItems, archiveRecordsFile)) {
+            return;
+        }
+
+        for (var batchItem : batchItems.getSuccessful()) {
+            dataStoreToArmHelper.updateExternalObjectDirectoryStatus(batchItem.getArmEod(), EodHelper.armDropZoneStatus(), userAccount);
+            logApi.armPushSuccessful(batchItem.getArmEod().getId());
+        }
+
+    }
+
+    private boolean writeManifestAndCopyToArm(UserAccountEntity userAccount, ArmBatchItems batchItems, File archiveRecordsFile) {
+        try {
+            if (!batchItems.getSuccessful().isEmpty()) {
+                dataStoreToArmHelper.writeManifestFile(batchItems, archiveRecordsFile);
+                updateObjectStateRecordManifestCreated(batchItems, archiveRecordsFile);
+                copyMetadataToArm(archiveRecordsFile);
+                updateObjectStateRecordManifestSuccessOrFailure(batchItems, archiveRecordsFile, true);
+            } else {
+                log.warn("No EODs were able to be processed, skipping manifest file creation");
+            }
+        } catch (Exception e) {
+            String errorMessage = String.format("Error during generation of DETS batch manifest file %s", archiveRecordsFile.getName());
+            log.error(errorMessage, e);
+            batchItems.getSuccessful().forEach(batchItem -> dataStoreToArmHelper.recoverByUpdatingEodToFailedArmStatus(batchItem, userAccount));
+            final String errorMessageWithStackTrace = errorMessage + " - " + ExceptionUtils.getStackTrace(e);
+            batchItems.getFailed().forEach(batchItem -> updateObjectStateRecordStatus(batchItem.getArmEod(), errorMessageWithStackTrace));
+            updateObjectStateRecordManifestSuccessOrFailure(batchItems, archiveRecordsFile, false);
```

Comment:
when ANY exception happens this method is:

- setFlagMfstTransfToArml(false)
- setting error message on OSR status to `"Manifest file creation failed"`.

but that exception could NOT be to related to the manifest file copy to ARM, it could have happened in a step before the copy. And so setting that "error status" on the OSR is a bit misleading I think